element_deeplabcut/dlc.py

```diff
@@ -142,154 +127,123 @@ class File(dj.Part):
-                          model_prefix: str = "", filter_type: str = "",
-                          track_method: str = "", scorer_legacy: bool = False,
-                          skip_duplicates=False):
-        param_dict = {'paramset_idx': paramset_idx, 'shuffle': shuffle,
-                      'train_fraction': train_fraction,  'filter_type': filter_type,
-                      'track_method': track_method, 'scorer_legacy': scorer_legacy}
+    def insert_new_params(cls, paramset_desc: str, params: dict, paramset_idx: int = None):
+        for required_param in cls.required_parameters:
+            assert required_param in params, f'Missing required parameter: {required_param}'
+
+        if paramset_idx is None:
+            paramset_idx = (dj.U().aggr(cls, n='max(paramset_idx)').fetch1('n') or 0) + 1
 
-        param_set_hash = dict_to_uuid(param_dict)
-        param_query = cls & {'param_set_hash': param_set_hash}
+        param_dict = {'paramset_idx': paramset_idx,
+                      'paramset_desc': paramset_desc,
+                      'params': params,
+                      'param_set_hash':  dict_to_uuid(params)
+                      }
+        param_query = cls & {'param_set_hash': param_dict['param_set_hash']}
 
         if param_query:  # If the specified param-set already exists
             existing_paramset_idx = param_query.fetch1('paramset_idx')
-            if skip_duplicates or existing_paramset_idx == paramset_idx:
-                return  # If the existing set has the same paramset_idx: job done
-            else:       # If not: human error adding same paramset with new index
+            if existing_paramset_idx == paramset_idx:  # If the existing set has the same paramset_idx: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
                 raise dj.DataJointError(
-                    'The specified parameter set already exists - paramset_idx: '
-                    + f'{existing_paramset_idx}')
+                    f'The specified param-set already exists'
+                    f' - with paramset_idx: {existing_paramset_idx}')
         else:
-            param_dict.update({'param_set_hash': param_set_hash})
-            cls.insert1(param_dict, skip_duplicates=skip_duplicates)
+            if {'paramset_idx': paramset_idx} in cls.proj():
+                raise dj.DataJointError(
+                    f'The specified paramset_idx {paramset_idx} already exists,'
+                    f' please pick a different one.')
+            cls.insert1(param_dict)
 
 
 @schema
 class TrainingTask(dj.Manual):
-    definition = """      # Info required to specify 1 model
+    definition = """      # Specification for a DLC model training instance
     -> VideoRecording     # labeled video for training
     -> ConfigParamSet
     training_id: int
+    ---
+    model_prefix='': varchar(32)

```

Comment:
prefix can be removed here and we can rely on it being present in the longblob, if provided